c++/src/Reader.cc

```diff
@@ -993,12 +995,25 @@ namespace orc {
     }
   }
 
+  int64_t getEpochOffset() {
+    struct tm epoch;
+    epoch.tm_sec = 0;
+    epoch.tm_min = 0;
+    epoch.tm_hour = 0;
+    epoch.tm_mday = 1;
+    epoch.tm_mon = 0;
+    epoch.tm_year = 2015 - 1900;
+    epoch.tm_isdst = 0;
+    return static_cast<int64_t>(mktime(&epoch));
+  }
+
```

Comment:
That is the representation of 1 Jan 2015 00:00:00, so it is converting that date into a time_t. Look at the man page for mktime.